lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -682,6 +683,12 @@ sub overview {
     my @latest_jobs = $self->schema->resultset('Jobs')->complex_query(%$search_args)->latest_jobs($until);
     ($stash{archs}, $stash{results}, $stash{aggregated}) = $self->prepare_job_results(\@latest_jobs);
 
+    # Allow n queries per minute, per user (if logged in)
+    my $lockname = 'webui_overview_rate_limit';
+    if (my $user = $self->current_user) { $lockname .= $user->username }
+    return $self->render(json => {error => 'Rate limit exceeded'}, status => 400)
```

Comment:
Why do this *after* querying the db for jobs? Shouldn't it be done before?

Reply:
It would also be nice to make this a helper instead of duplicating code from the search.

Reply:
Agreed, definitely should be a helper.